src/main/java/hudson/security/AuthorizationMatrixProperty.java

```diff
@@ -312,6 +342,13 @@ public Object unmarshal(HierarchicalStreamReader reader,
 			    reader.moveDown();
 			    as.setBlocksInheritance("true".equals(reader.getValue()));
 			    reader.moveUp();
+                            prop = reader.peekNextChild(); // We check the next field
+			}
+
+			if("blocksParentInheritance".equals(prop)) {
```

Comment:
Such approach presumes strict order of fields. It is generally OK, but having the logic within `while` iterator would be preferable. Not an issue in this PR